src/sentry/tasks/store.py

```diff
@@ -210,79 +214,76 @@ def _do_symbolicate_event(cache_key, start_time, event_id, symbolicate_task, dat
-            )
-            retry_symbolicate_event.apply_async(
-                args=(),
-                kwargs={
-                    "symbolicate_task_name": symbolicate_task.__name__,
-                    "task_kwargs": {
-                        "cache_key": cache_key,
-                        "event_id": event_id,
-                        "start_time": start_time,
-                    },
-                },
-                countdown=e.retry_after,
-            )
-            return
-    except Exception:
-        metrics.incr(
-            "tasks.store.symbolicate_event.fatal",
-            tags={"reason": "error", "symbolication_function": symbolication_function.__name__},
-        )
-        error_logger.exception("tasks.store.symbolicate_event.symbolication")
-        data.setdefault("_metrics", {})["flag.processing.error"] = True
-        data.setdefault("_metrics", {})["flag.processing.fatal"] = True
-        has_changed = True
+                    break
+                except RetrySymbolication as e:
+                    if (
+                        start_time
+                        and (time() - start_time) > settings.SYMBOLICATOR_PROCESS_EVENT_WARN_TIMEOUT
+                    ):
+                        error_logger.warning(
+                            "symbolicate.slow",
+                            extra={"project_id": project_id, "event_id": event_id},
+                        )
+                    if (
+                        start_time
+                        and (time() - start_time) > settings.SYMBOLICATOR_PROCESS_EVENT_HARD_TIMEOUT
+                    ):
+                        # Do not drop event but actually continue with rest of pipeline
+                        # (persisting unsymbolicated event)
+                        metrics.incr(
+                            "tasks.store.symbolicate_event.fatal",
+                            tags={
+                                "reason": "timeout",
+                                "symbolication_function": symbolication_function.__name__,
+                            },
+                        )
+                        error_logger.exception(
+                            "symbolicate.failed.infinite_retry",
+                            extra={"project_id": project_id, "event_id": event_id},
+                        )
+                        data.setdefault("_metrics", {})["flag.processing.error"] = True
+                        data.setdefault("_metrics", {})["flag.processing.fatal"] = True
+                        has_changed = True
+                        break
+                    else:
+                        # sleep for `retry_after` seconds and try again
+                        metrics.incr(
+                            "tasks.store.symbolicate_event.retry",
+                            tags={"symbolication_function": symbolication_function.__name__},
+                        )
+                        sleep(e.retry_after)
```

Comment:
I'm not sure it's a good idea to always respect the `retry_after` atm it seems to be hardcoded in symbolicator to 30s. https://github.com/getsentry/symbolicator/blob/5fd8b3d4c0b5299fecd5864e171b34df7ba80c4c/src/actors/symbolication.rs#L234

Reply:
Introduced a new setting that will max it to 5s regardless of what comes from symbolicator. This might be revisited once symbolicator implements more sophisticated estimations.